app/assets/javascripts/utilities/showUserAlertModal.js

```diff
@@ -30,24 +30,67 @@ function showUserAlertModal(title, text, confirm_text) {
  * Displays a user rate limit alert modal letting the user know what they did that exceeded a rate limit,
  * and gives them links to explain why they must wait
  *
- * @function showUserAlertModal
+ * @function showRateLimitModal
  * @param {string} action_text Description of the action taken by the user
  * @param {string} next_action_text Description of the next action that can be taken
  *
  * @example
  * showRateLimitModal('Made a comment', 'comment again')
  */
-function showRateLimitModal(action_text, next_action_text) {
-  let rateLimitText = buildRateLimitText(action_text, next_action_text);
+function showRateLimitModal(
+  action_text,
+  next_action_text,
+  timeframe = 'a moment',
+) {
+  let rateLimitText = buildRateLimitText(
+    action_text,
+    next_action_text,
+    timeframe,
+  );
   let rateLimitLink = '/faq';
   showUserAlertModal(
-    'Wait a moment...',
+    `Wait ${timeframe}...`,
     rateLimitText,
     'Got it',
     rateLimitLink,
     'Why do I have to wait?',
   );
 }
+/**
+ * Displays the corresponding modal after an error.
+ *
+ * @function showModalAfterError
+ * @param {Object} response The response from the API
+ * @param {string} action_text Description of the action taken by the user
+ * @param {string} next_action_text Description of the next action that can be taken
+ * @param {string} timeframe Description of the time that we need to wait
+ *
+ * @example
+ * showModalAfterError(response, 'made a comment', 'making another comment', 'a moment');
+ */
+function showModalAfterError(
+  response,
+  action_text,
+  next_action_text,
+  timeframe,
+) {
```

Comment:
```suggestion
function showModalAfterError({
  response,
  actionText,
  nextActionText,
  timeframe,
}) {
```
refactor (blocking) - with so many props it will be easier for developer experience to pass them as an object, so we can then call like e.g. 

```js
showModalAfterError({ 
  response, 
  actionText: 'uploaded an image', 
  nextActionText: 'uploading another image', 
  timeframe: 'a moment'})
```